src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -704,6 +702,42 @@ public async Task TestSqlSettingPostStartupConfigurations(string configurationEn
             Assert.AreEqual(HttpStatusCode.BadRequest, postConfigOpenApiSwaggerEndpointAvailability.StatusCode);
         }
 
+        /// <summary>
+        /// Tests that sending configuration to the DAB engine post-startup will properly hydrate even with data-source-files specified.
+        /// </summary>
+        [TestCategory(TestCategory.MSSQL)]
+        [TestMethod("Validates RuntimeConfig setup for post-configuraiton hydration with datasource-files specified.")]
+        [DataRow(CONFIGURATION_ENDPOINT)]
+        [DataRow(CONFIGURATION_ENDPOINT_V2)]
+        public async Task TestValidMultiSourceRunTimePostStartupConfigurations(string configurationEndpoint)
+        {
+            TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+            HttpClient httpClient = server.CreateClient();
+
+            RuntimeConfig config = AuthorizationHelpers.InitRuntimeConfig(
+                entityName: POST_STARTUP_CONFIG_ENTITY,
+                entitySource: POST_STARTUP_CONFIG_ENTITY_SOURCE,
+                roleName: POST_STARTUP_CONFIG_ROLE,
+                operation: EntityActionOperation.Read,
+                includedCols: new HashSet<string>() { "*" });
+
+            // Set up Configuration with DataSource files.
+            config = config with { DataSourceFiles = new DataSourceFiles(new List<String>() { "file1", "file2" }) };
+
+            JsonContent content = GetPostStartupConfigParams(MSSQL_ENVIRONMENT, config, configurationEndpoint);
+
+            HttpResponseMessage postResult = await httpClient.PostAsync(configurationEndpoint, content);
+            Assert.AreEqual(HttpStatusCode.OK, postResult.StatusCode);
+
+            RuntimeConfigProvider configProvider = server.Services.GetService<RuntimeConfigProvider>();
+
+            Assert.IsNotNull(configProvider, "Configuration Provider shouldn't be null after setting the configuration at runtime.");
+            Assert.IsTrue(configProvider.TryGetConfig(out RuntimeConfig configuration), "TryGetConfig should return true when the config is set.");
+            Assert.IsNotNull(configuration, "Config returned should not be null.");
+
+            Assert.AreEqual(1, configuration.GetDataSourceNamesToDataSourcesIterator().Count(), "Only the base datasource should be used in case of late hydration of config.");
```

Comment:
Does order matter? is the base datasource the datasource that is listed first?

Reply:
reworded this message and modifies the assert. I guess it was confusing. What the message meant to say was - in late hydration scenario like static web apps you have the base config present.

Reply:
base config present meaning that the other datasources are not read? e.g. you have
> `config = config with { DataSourceFiles = new DataSourceFiles(new List<String>() { "file1", "file2" }) };`

Because this test does not test config validation, it just tests that the runtimeconfig object was successfully created after deserializing the config -> would it still be valid to assert that the count of datasources is just 1 (which is the base config)?

Reply:
Thats fair, will add that assert back.